cwms-data-api/src/main/java/cwms/cda/api/watersupply/WaterPumpDeleteController.java

```
@@ -0,0 +1,147 @@
+            WaterContractDao contractDao = getContractDao(dsl);
+            CwmsId projectLocation = buildCwmsId(officeId, projectName);
+            List<WaterUserContract> contract = contractDao.getAllWaterContracts(projectLocation, entityName);
+
+            for (WaterUserContract waterUserContract : contract) {
+                if (waterUserContract.getContractId().getName().equals(contractName)) {
+                    switch (pumpType) {
+                        case IN:
+                            contractDao.removePumpFromContract(waterUserContract,
+                                    waterUserContract.getPumpInLocation().getPumpLocation().getName(),
+                                    PumpType.IN, deleteAccounting);
+                            ctx.status(HttpServletResponse.SC_NO_CONTENT);
+                            return;
+                        case OUT:
+                            contractDao.removePumpFromContract(waterUserContract,
+                                    waterUserContract.getPumpOutLocation().getPumpLocation().getName(),
+                                    PumpType.OUT, deleteAccounting);
+                            ctx.status(HttpServletResponse.SC_NO_CONTENT);
+                            return;
+                        case BELOW:
+                            contractDao.removePumpFromContract(waterUserContract,
+                                    waterUserContract.getPumpOutBelowLocation().getPumpLocation().getName(),
+                                    PumpType.BELOW, deleteAccounting);
+                            ctx.status(HttpServletResponse.SC_NO_CONTENT);
+                            return;
+                        default:
+                            CdaError error = new CdaError("Invalid pump type provided.");
+                            LOGGER.log(Level.SEVERE, "Invalid pump type provided.");
+                            ctx.status(HttpServletResponse.SC_NOT_FOUND).json(error);
```

This should return an error code 400 as it is bad input by the client. You should be able to throw an IllegalArgumentException here and it'll get handled appropriately.

---

Added a throw statement and removed manual error handling